components/Dialogs/src/private/MDCAlertActionManager.m

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCAlertActionManager.h"
+
+@interface MDCAlertActionManager ()
+@property(nonatomic, nonnull, strong) NSMapTable *actionButtons;
+@end
+
+@implementation MDCAlertActionManager {
+  NSMutableArray<MDCAlertAction *> *_actions;
+}
+
+@dynamic sortedButtons;
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _actions = [[NSMutableArray alloc] init];
+    _actionButtons = [NSMapTable mapTableWithKeyOptions:NSMapTableWeakMemory
+                                           valueOptions:NSMapTableStrongMemory];
+  }
+  return self;
+}
+
+- (NSArray<MDCButton *> *)sortedButtons {
```

Comment:
The name implies that we're applying a sorting algorithm to the buttons, but it looks like we may be retrieving the buttons in action order. Consider `allButtonsInActionOrder`.

Reply:
I've renamed it, although this may require further consideration. Should this method create buttons that are missing, similar to "buttonForAction"?  I've renamed it, but didn't addressing the larger question in the PR. I moved the discussion to the followup issue #5447.